localflavor/nl/models.py

```diff
@@ -117,9 +119,15 @@ class NLBankAccountNumberField(models.CharField):
     This model field uses :class:`validators.NLBankAccountNumberFieldValidator` for validation.
 
     .. versionadded:: 1.1
+
+    .. deprecated:: 1.6
+        Use `localflavor.generic.models.IBANField` with included_countries=('nl') option instead.
     """
 
     def __init__(self, *args, **kwargs):
+        warnings.warn("NLBankAccountNumberField is deprecated. Use `localflavor.generic.models.IBANField` "
+                      "with included_countries=('nl') option instead.", RemovedInLocalflavor20Warning)
+
```

Comment:
For deprecating the phone number models we're using the system check framework. Do you think it make sense to use the system check framework here as well?
https://github.com/django/django-localflavor/blob/master/localflavor/deprecation.py#L21-L24

Also, `IBANField` isn't a direct replacement for `NLBankAccountNumberField`. Users would need to do a data migration and generate the IBAN for each NL bank account number. My feeling is that adding this data migration is out of scope for localflavor. Calculating the check digits straightforward but we'd need to somehow map the NL account numbers to a 4 character IBAN bank ID and I don't this mapping readily available. I think we should just have a note stating that a data migration is required if users want to directly replace the  `NLBankAccountNumberField` with the `IBANFeild`. Otherwise, user will need to use both fields in parallel for a while.

Reply:
I have no preference regarding the systems check framework vs using `warnings.warn`. If you do prefer using the systems check framework, I'll be happy to change the patch.

You're right about the data migration, I'll add a note.

Reply:
I prefer the systems check framework to keep things consistent with the other model deprecations. Thanks.